components/core/src/clp_s/clp-s.cpp

```diff
@@ -113,9 +123,15 @@ bool compress(CommandLineArguments const& command_line_arguments) {
     }
 
     clp_s::JsonParser parser(option);
```

Comment:
We might want to include the file type information in JsonParserOption as part of this PR -- e.g. have an enum field `file_type` that borrows the enum we'll add in CommandLineArguments.